iOS/WAStickersThirdParty/UIAlertController+Additions.swift

```diff
@@ -22,14 +22,41 @@ extension UIAlertController {
         activity.startAnimating()
     }
 
-    func addImageView(withImage image: UIImage) {
+    func addImageView(withImage image: UIImage, animated: Bool) {
         var stickerImageViewLength: CGFloat = 100.0
         if #available(iOS 9.0, *) {
             stickerImageViewLength = 125
         }
 
         let stickerImageView: UIImageView = UIImageView(image: image)
         stickerImageView.translatesAutoresizingMaskIntoConstraints = false
+
+        if animated {
+            if let images = image.images {
+                stickerImageView.animationImages = images
+                stickerImageView.animationDuration = image.duration
+
+                stickerImageView.layer.speed = 1.0
+                stickerImageView.layer.timeOffset = 0.0
+                stickerImageView.layer.beginTime = 0.0
+
+                let animation = QuartzCore.CAKeyframeAnimation(keyPath:"contents")
+                var values: [CGImage] = []
+                for image in images {
+                    guard let cgImage = image.cgImage else {
+                        continue
+                    }
+                    values.append(cgImage)
+                }
```

Comment:
A good way to shorten this is to use compactMap. It will also automatically get rid of nil values.

`let values: [CGImage] = images.compactMap { $0.cgImage }`